Lib/ipaddress.py

```diff
@@ -594,6 +594,76 @@ def _get_address_key(self):
     def __reduce__(self):
         return self.__class__, (self._ip,)
 
+    def __format__(self, fmt):
+        """Returns a formatted string. Supported presentation types are:
+            's': returns the IP address as a string (default)
+            'b' or 'n': converts to binary and returns a zero-padded string
+            'X' or 'x': converts to upper- or lower-case hex and returns a zero-padded string
+
+            For binary and hex presentation types, the alternate form specifier
+            '#' and the grouping option '_' are supported.
```

Comment:
Please follow PEP 257 for docstrings.

Reply:
I looked at the string.format() docstring as an example of how to do docstrings right. It is:

`S.format(*args, **kwargs) -> str

`Return a formatted version of S, using substitutions from args and kwargs.
The substitutions are identified by braces ('{' and '}').`

However, PEP257 says
`The one-line docstring should NOT be a "signature" reiterating the function/method parameters`

Should I follow the string.format() docstring and give a signature as the first line, or follow PEP257 and give a sentence, along the lines of "Returns an IP address as a formatted string."?

Not trying to language-lawyer, I just want to make sure I do the right thing here.

Reply:
I pushed a proposal out, it seems OK, let me know if  you'd prefer changes.

Reply:
Follow 257 more than str.format(); the latter is implemented in C and follows a scheme that has fallen out of favor, but we haven't been able to fix it due to not having support for `*args, **kwargs` in Argument Clinic.

This looks good, other than that all but the first line should be dedented 4 spaces (indented a total of 8 to match the rest of the function body).

Reply:
Thanks!  I dedented the docstring, so as far as I know I've addressed all outstanding comments.